### PR TITLE
Use get url

### PR DESCRIPTION
**Pull request by brunoleon**
---
name: Pull request
about: Use get URL to download Tomcat archive

---

**Describe the change**
Unarchive does no use system PROXY and thuis breaks installation when you must use one.
get_url does use system proxy, so I splitted up download/unarchive in two parts so that unarchive refers to the downloaded archive from get_rul.

**Testing**
Just use the module as usual, there should not be any difference.

### CODE REVIEW

**Comment by robertdebock** [comment]
Okay, cool. did not know about unarchive not supporting proxies. Also like the `lib` addition.

Building fails though. Do you need help or can you fix the failing builds yourself?

**Comment by robertdebock** [comment]
Hey, the build is failing on RHEL-6:

```
    TASK [ansible-role-tomcat : download tomcat source] ****************************
    fatal: [tomcat-centos-6]: FAILED! => {"changed": false, "msg": "Failed to validate the SSL certificate for www-eu.apache.org:443. Make sure your managed systems have a valid CA certificate installed. If the website serving the url uses SNI you need python >= 2.7.9 on your managed machine  (the python executable used (/usr/bin/python) is version: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37) [GCC 4.4.7 20120313 (Red Hat 4.4.7-17)]) or you can install the `urllib3`, `pyOpenSSL`, `ndg-httpsclient`, and `pyasn1` python modules to perform SNI verification in python >= 2.6. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible. The exception msg was: hostname 'www-eu.apache.org' doesn't match either of '*.openoffice.org', 'openoffice.org'."}
```

There is a variable `{{ tomcat_validate_certs }}` that you can use with the parameter `validate_certs` for the `get_url` module.

Note; RHEL8 has just been announces, I'm going to drop support for RHEL6 soon, so if this become too difficult, just remove the CentOS-6 tests from `.travis.yml`.

File: tasks/instance.yml [hunk 1]
```diff
@@ -19,9 +19,15 @@
     owner: "{{ instance.user | default(tomcat_user) }}"
     group: "{{ instance.group | default(tomcat_group) }}"
 
+- name: download tomcat source
+  get_url:
+    url: "{{ tomcat_unarchive_url }}"
+    dest: "{{ tomcat_directory }}/{{ instance.name }}/tomcat_src.tar.gz"
+    mode: 0440
+
```
Comment (robertdebock) [inline tasks/instance.yml]: This is where the `validate_certs: "{{ tomcat_validate_certs }}"` should go, line 27 or so.

File: tasks/lib.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+- name: deploy lib
+  get_url:
+    url: "{{ lib.url }}"
+    dest: "{{ tomcat_directory }}//{{ instance.name }}/lib/"
+    validate_certs: no
```
Comment (robertdebock) [inline tasks/lib.yml]: Here too, please use `{{ tomcat_validate_certs }}`.

**Comment by brunoleon** [comment]
I updated the pull request

Example lib:
    libs:
      - url: "https://search.maven.org/remotecontent?filepath=io/prometheus/simpleclient/0.6.0/simpleclient-0.6.0.jar"

**Comment by robertdebock** [comment]
Look good, a few build steps failed, I retried them. Should be ready to merge in a moment.

**Comment by brunoleon** [comment]
Thank you !

**Pull request merged**